cli/azd/pkg/project/importer.go

```diff
@@ -92,11 +92,122 @@ func (im *ImportManager) ServiceStable(ctx context.Context, projectConfig *Proje
+func (im *ImportManager) sortServicesByDependencies(
+	services []*ServiceConfig,
+	projectConfig *ProjectConfig,
+) ([]*ServiceConfig, error) {
+	// Validate dependencies exist
+	if err := im.validateServiceDependencies(services, projectConfig); err != nil {
+		return nil, err
+	}
+
+	// Check if any service has dependencies
+	hasDependencies := false
+	for _, svc := range services {
+		if len(svc.Uses) > 0 {
+			hasDependencies = true
+			break
+		}
+	}
+
+	// If no dependencies are defined, maintain the original alphabetical ordering
+	if !hasDependencies {
+		slices.SortFunc(services, func(x, y *ServiceConfig) int {
+			return strings.Compare(x.Name, y.Name)
+		})
+		return services, nil
+	}
+
+	// Create dependency graph
+	graph := make(map[string][]string)
+	inDegree := make(map[string]int)
+
+	// Initialize graph and in-degree count
+	for _, svc := range services {
+		graph[svc.Name] = []string{}
+		inDegree[svc.Name] = 0
+	}
+
+	// Build dependency edges
+	for _, svc := range services {
+		for _, dependency := range svc.Uses {
+			// Only consider service-to-service dependencies for ordering
+			// (service-to-resource dependencies are declarative only)
+			if _, isService := graph[dependency]; isService {
+				graph[dependency] = append(graph[dependency], svc.Name)
+				inDegree[svc.Name]++
+			}
+		}
+	}
+
+	// Topological sort using Kahn's algorithm
+	var result []*ServiceConfig
+	var queue []string
+
+	// Find all services with no dependencies
+	for serviceName, degree := range inDegree {
+		if degree == 0 {
+			queue = append(queue, serviceName)
+		}
+	}
+
+	// Process services in dependency order
+	serviceMap := make(map[string]*ServiceConfig)
+	for _, svc := range services {
+		serviceMap[svc.Name] = svc
+	}
 
-	return allServicesSlice, nil
+	for len(queue) > 0 {
+		// Remove a service with no dependencies
+		current := queue[0]
+		queue = queue[1:]
+		result = append(result, serviceMap[current])
+
+		// Update dependencies of services that depend on current service
+		for _, dependent := range graph[current] {
+			inDegree[dependent]--
+			if inDegree[dependent] == 0 {
+				queue = append(queue, dependent)
+			}
+		}
+	}
+
+	// Check for circular dependencies
+	if len(result) != len(services) {
+		return nil, fmt.Errorf("circular dependency detected in services")
+	}
+
+	return result, nil
+}
+
+// validateServiceDependencies ensures all dependencies referenced in service Uses exist
+func (im *ImportManager) validateServiceDependencies(services []*ServiceConfig, projectConfig *ProjectConfig) error {
+	serviceNames := make(map[string]bool)
+	for _, svc := range services {
+		serviceNames[svc.Name] = true
+	}
+
+	for _, svc := range services {
+		for _, dependency := range svc.Uses {
+			// Check if dependency exists as a service or resource
+			if !serviceNames[dependency] && (projectConfig.Resources == nil || projectConfig.Resources[dependency] == nil) {
```

Comment:
should there be a condition to catch if `svc.Name == dependency`? and error as service can't use itself?